### PR TITLE
:sparkles: IdentityForm: Username validation

### PR DESCRIPTION
**Pull request by gildub**
Simplifies create/update modal
Jira username requires valid email

Resolves https://github.com/konveyor/tackle2-ui/issues/858

### CODE REVIEW

**Comment by ibolton336** [comment]
Nice tech debt fixes here!

File: client/src/app/pages/identities/identities.tsx [hunk 1]
```diff
@@ -173,35 +171,9 @@ export const Identities: React.FC = () => {
...
-    }
-
-    closeIdentityModal();
-  };
-
-  const handleOnIdentityUpdated = () => {
-    setRowToUpdate(undefined);
+  const closeCreateUpdateModal = () => {
+    setCreateUpdateModalState(null);
+    refetch;
```
Comment (ibolton336) [inline client/src/app/pages/identities/identities.tsx:176]: Is this supposed to be called?

**Review by ibolton336 — CHANGES_REQUESTED** [review]

File: client/src/app/pages/identities/identities.tsx [hunk 2]
```diff
@@ -317,16 +292,27 @@ export const Identities: React.FC = () => {
...
+                })
+          }
+          variant={ModalVariant.medium}
+          isOpen={isCreateUpdateModalOpen}
+          onClose={closeCreateUpdateModal}
+        >
+          <IdentityForm
+            identity={identityToUpdate ? identityToUpdate : undefined}
+            onClose={closeCreateUpdateModal}
+            xmlValidator={validateXML}
```
Comment (ibolton336) [inline client/src/app/pages/identities/identities.tsx:313]: Not needed

File: client/src/app/pages/identities/identities.tsx [hunk 3]
```diff
@@ -47,6 +46,8 @@ import {
 } from "@app/queries/identities";
 import { useFetchApplications } from "@app/queries/applications";
 import { NotificationsContext } from "@app/shared/notifications-context";
+import { IdentityForm } from "./components/identity-form";
+import { validateXML } from "./components/identity-form/validateXML";
```
Comment (ibolton336) [inline client/src/app/pages/identities/identities.tsx:50]: Not needed

**Review by ibolton336 — CHANGES_REQUESTED** [review]

**Review by ibolton336 — CHANGES_REQUESTED** [review]
See inline.

File: client/src/app/pages/identities/identities.tsx (hunk 1 above)
Comment (gildub) [reply client/src/app/pages/identities/identities.tsx:176]: @ibolton336, normally we don't need it with `invalidateQueries()`
Let me double check...

Comment (ibolton336) [reply client/src/app/pages/identities/identities.tsx:176]: Cool - just saying if we do it needs perens 
```suggestion
    refetch();
```

File: client/src/app/pages/identities/identities.tsx (hunk 2 above)
Comment (ibolton336) [reply client/src/app/pages/identities/identities.tsx:313]: I am sorry - we may still need this. Need to check with Jeff/Ramon here about support for mvn identities for mta 7. I Know we will need it for v0.2.0

File: client/src/app/pages/identities/identities.tsx (hunk 3 above)
Comment (ibolton336) [reply client/src/app/pages/identities/identities.tsx:50]: I am sorry - we may still need this. Need to check with Jeff/Ramon here about support for mvn identities for mta 7. I Know we will need it for v0.2.0

File: client/src/app/pages/identities/identities.tsx (hunk 2 above)
Comment (gildub) [reply client/src/app/pages/identities/identities.tsx:313]: @ibolton336, this is out of scope for this PR. I haven't changed anything from previous modal. So let's open another issue if needed.

File: client/src/app/pages/identities/identities.tsx (hunk 1 above)
Comment (gildub) [reply client/src/app/pages/identities/identities.tsx]: @ibolton336, redundant, I removed it.

**Comment by ibolton336** [comment]
@gildub Did we confirm we want to constrain the jira cred username to an email?

**Review by ibolton336 — CHANGES_REQUESTED** [review]
One question re: email vs string for jira instance creds.

**Comment by gildub** [comment]
@ibolton336, yes Jira auth is based on Jira email for associated account.

**Comment by ibolton336** [comment]
@gildub If it is always email, we should change the field name to email.

**Comment by gildub** [comment]
@ibolton336, well Jira users know a username is an email address.
It's like most of user accounts on the web, isn't ?
That said we could display `"Username (email)"`.
Not a blocker

**Comment by ibolton336** [comment]
> @ibolton336, well Jira users know a username is an email address. It's like most of user accounts on the web, isn't ? That said we could display `"Username (email)"`. Not a blocker

Last I checked, the username differed based on the type of JIRA instance used. Is that no longer the case? The credential will fail if they use their jira username so I would think we would want to let the user know that the email is required rather than the jira user name.

**Comment by ibolton336** [comment]
> @ibolton336, well Jira users know a username is an email address. It's like most of user accounts on the web, isn't ? That said we could display `"Username (email)"`. Not a blocker

Looks like github accepts either username or email. So if JIRA operates this way, I would think something needs to change with how we display the field. 


Screenshot 2023-05-24 at 3 07 29 PM

**Comment by gildub** [comment]
Not sure about earlier generalization I made as this is semantic.

Meanwhile I based my approach from Jira (documentation says)[https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/] says about creating an API token: "You can make HTTP basic authentication requests with your email address as the username and this API token as the password." 

I'm happy to use either "Username" or "Email" or both. We're likely need validation from UXD.

File: client/src/app/pages/identities/components/identity-form/identity-form.tsx [hunk 4]
```diff
@@ -645,7 +645,7 @@ export const IdentityForm: React.FC<IdentityFormProps> = ({
           <HookFormPFTextInput
             control={control}
             name="user"
-            label="Username"
+            label={values.kind === "jira" ? "Username's email" : "Username"}
```
Comment (ibolton336) [inline client/src/app/pages/identities/components/identity-form/identity-form.tsx:648]: Per @justinxhale from UXD:

"Jira username or email”

**Pull request merged**